lib/utils.js

```diff
@@ -132,7 +132,9 @@ exports.compact = function (obj, references) {
         var compacted = [];
 
         for (var i = 0; i < obj.length; ++i) {
-            if (typeof obj[i] !== 'undefined') {
+            if (Object.prototype.toString.call(obj[i]) === '[object Object]') {
```

Comment:
this test won't give true for arrays or functions - perhaps this should test for any non-primitive?